### PR TITLE
Better clearing of session messages after display

### PR DESCRIPTION
**Pull request by kyrena**
### Description

Rarely, I viewed flash notices two times on the same page, because multiple blocks call `<?php echo $this->getMessagesBlock()->getGroupedHtml() ?>`.

So this PR prevent that.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All automated tests passed successfully (all builds are green)
 - [ ] Add yourself to contributors list

### CODE REVIEW

File: app/code/core/Mage/Core/Block/Messages.php [hunk 1]
```diff
@@ -247,6 +247,7 @@ public function getGroupedHtml()
         if ($html) {
             $html .= '</' . $this->_messagesFirstLevelTagName . '>';
         }
+        $this->_messages = null;
```
Comment (ADDISON74) [inline app/code/core/Mage/Core/Block/Messages.php:250]: PHPStan is not happy with this statement.

**Review by ADDISON74 — APPROVED** [review]

File: app/code/core/Mage/Core/Block/Messages.php (hunk 1 above)
Comment (ADDISON74) [reply app/code/core/Mage/Core/Block/Messages.php:250]: In my case I push the wrong button and I approved this PR. It needs more 2 green approves.

The reported issue I encountered a few times over the years.

Comment (fballiano) [inline app/code/core/Mage/Core/Block/Messages.php:250]:
```suggestion
        $this->getMessageCollection()->clear();
```

Comment (kyrena) [reply app/code/core/Mage/Core/Block/Messages.php:250]: Yes, I don't remember if I tested or not this solution before.
Working for now, I continue testing it.

File: app/code/core/Mage/Core/Block/Messages.php [hunk 2]
```diff
@@ -247,6 +247,7 @@ public function getGroupedHtml()
         if ($html) {
             $html .= '</' . $this->_messagesFirstLevelTagName . '>';
         }
+        $this->getMessageCollection()->clear();
```
Comment (fballiano) [inline app/code/core/Mage/Core/Block/Messages.php:250]:
```suggestion
        $this->_messages = $this->getMessageCollection()->clear();
```

**Review by fballiano — APPROVED** [review]

**Pull request merged**